nix/codex.nix

```diff
@@ -0,0 +1,58 @@
+{ config, lib, pkgs, ... }:
+
+let
+  inherit (lib)
+    types mkEnableOption mkOption mkIf literalExpression;
+  
+  toml = pkgs.formats.toml { };
+
+  cfg = config.services.codex;
+in
+{
+  options = {
+    services.codex = {
+      enable = mkEnableOption "Codex Node service.";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.callPackage ./default.nix { };
+        defaultText = literalExpression "pkgs.codex";
+        description = lib.mdDoc "Package to use as Codex node.";
+      };
+
+      configFile = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "Path to the Codex configuration file.";
+      };
+
+      configObject = lib.mkOption {
+        default = { };
+        type = toml.type;
+        description = ''Structured settings object that will be used to generate a TOML config file.'';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.etc = {
+      "codex/config.toml".source = toml.generate "config.toml" cfg.configObject;
+    };
```

Comment:
If you want to do it like this, you will need to include `restartTriggers` for the service, because right now the service would not be restarted if the config file was changed. Normally NixOS services reference the generated config directly from `/nix/store`, which means the service would be restarted every time it changes, since the path would be different and service definition would have to be changed.